pkg/archive/spec.go

```diff
@@ -1,15 +1,171 @@
 package archive
 
+import (
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+var (
+	alphanumeric = regexp.MustCompile(`[^a-zA-Z0-9]+`)
+
+	ExtensionMap = map[string]string{
```

Comment:
These could probably be done on `targets.Target` to keep things in one place.
So, e.g. `targets.Bullseye()` would have a value set to get the extension name, os name, and version string.

Reply:
`targets.Bullseye()` requires a context and a dagger client as the 1st and 2nd args, and these maps are only being used by the utility programs (which don't need either of those). How about static structs in the `targets` package that are themselves used as a member of the `Target` struct?

Reply:
> `targets.Bullseye()` requires a context and a dagger client as the 1st and 2nd args, and these maps are only being used by the utility programs (which don't need a client). How about static structs in the `targets` package that are themselves used as a member of the `Target` struct?

Actually, this is a problem because `targets` already imports `archive`, so we can't have `archive` also import `targets`.